### PR TITLE
Refactor command line parameters + Improve error messages

### LINKED ISSUE #17: Improve command line parameters
We use command line parameters in both electron and backend. Therefore, it makes sense to have a complete strategy on how these are implemented.

Additionally:

- accept port and host as command line argument to start backend server
- accept command line parameter to use an in memory database

### PR DESCRIPTION
**Pull request by martingraf4**
Closes #17 

- created a combined configuration file for command line parameters
  - all command line parameters are passed to the api process
- improved some error messages

### CODE REVIEW

File: wrapper/src/api/tools/cli.ts [hunk 1]
```diff
@@ -1,22 +1,63 @@
 import * as dashdash from 'dashdash';
+import path from 'path';
 
-interface IArgs {
-  path?: string;
+interface IArgs extends dashdash.Results {
```
Comment (Lasklu) [inline wrapper/src/api/tools/cli.ts:4]: In every other file of the api, we do not use "I" as a prefix for interfaces AFAIK. Why is this needed here?

File: wrapper/src/api/providers/algorithm/algorithmProvider.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ export class AlgorithmProvider extends BaseAlgorithmProvider {
         )
         .join(', ');
       throw new Error(
-        `The algorithm ${algorithm.name} (${algorithm.id}) is used by the experiments ${experiments}.`
+        `The matching solution ${algorithm.name} (${algorithm.id}) is used by the experiments ${experiments}.`
```
Comment (Lasklu) [inline wrapper/src/api/providers/algorithm/algorithmProvider.ts]: 🥇

File: wrapper/src/api/tools/cli.ts (hunk 1 above)
Comment (martingraf4) [reply wrapper/src/api/tools/cli.ts]: done
